src/main/java/spark/Service.java

```diff
@@ -71,6 +73,7 @@
     protected Optional<Integer> webSocketIdleTimeoutMillis = Optional.empty();
 
     protected EmbeddedServer server;
+    protected Stack<String> pathStack = new Stack<>();
```

Comment:
I think `Stack` is one of those "not recommended" classes, quoting:
> A more complete and consistent set of LIFO stack operations is provided by the Deque interface and its implementations, which should be used in preference to this class. 

Stack is, IIRC, synchronized and eg. `ArrayDeque` probably not, but from what I understood (took a quick glance only, though) there is no need for it here. And if there was, using `Stack` wouldn't really close the window of opportunity for a race condition.